arm/Microsoft.Automation/automationAccounts/deploy.bicep

```diff
@@ -124,71 +105,75 @@ module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
 }
 
 resource automationAccount 'Microsoft.Automation/automationAccounts@2020-01-13-preview' = {
-  name: automationAccountName
+  name: name
   location: location
   tags: tags
   properties: {
     sku: {
       name: skuName
     }
   }
+}
 
-  resource automationAccount_modules 'modules@2020-01-13-preview' = [for (module, index) in modules: {
+module automationAccount_modules './modules/deploy.bicep' = [for (module, index) in modules: {
+  name: '${uniqueString(deployment().name, location)}-AutoAccount-Module-${index}'
```

Comment:
Design question: For storage account I removed all refernce to the parent resource type (here `-AutoAccount-`) to stay as far away from the char-limit as possible. Do you think we should find a solution that works for all, or do it on a case by case bases? Here we won't run into the same problems as with the storage account :D

Reply:
If we use the `uniqueString` we loose info about the main resource being deployed so probably `-AutoAccount-` is useful to keep.

Reply:
ok, then I'd propose do keep it on a case by case basis. For resources with many levels of nesting this will not work, but that does not mean we should go for the 'workaround' for all others

Reply:
Opened issue #472 to align all resources to the same convention